src/mqtt_client_adapter.cpp

```diff
@@ -88,6 +88,41 @@ static void update_erd(i_mqtt_client_t* _self, tiny_erd_t erd, const void* _valu
   self->client->publish(topic.c_str(), payload.c_str(), true);
 }
 
+static void update_erd_write_result(i_mqtt_client_t* _self, tiny_erd_t erd, bool success, tiny_erd_client_write_failure_reason_t failure_reason)
+{
+  auto self = reinterpret_cast<mqtt_client_adapter_t*>(_self);
+
+  auto erd_string = String(erd, HEX);
+  while(erd_string.length() < 4) {
+    erd_string = "0" + erd_string;
+  }
+
+  auto topic = String("geappliances/") + self->device_id + "/erd/0x" + erd_string + "/write_result";
+
+  if(success) {
+    self->client->publish(topic.c_str(), "success", true);
+  }
+  else {
+    switch(failure_reason) {
+      case tiny_erd_client_write_failure_reason_retries_exhausted:
+        self->client->publish(topic.c_str(), "retries exhausted", true);
+        break;
+
+      case tiny_erd_client_write_failure_reason_not_supported:
+        self->client->publish(topic.c_str(), "not supported", true);
+        break;
+
+      case tiny_erd_client_write_failure_reason_incorrect_size:
+        self->client->publish(topic.c_str(), "incorrect size", true);
+        break;
+
+      default:
+        self->client->publish(topic.c_str(), "incorrect size", true);
```

Comment:
Should we use a different string here?

Reply:
Whoops, this was supposed to be `"unknown"`